tf2_ros/src/transform_listener.cpp

```diff
@@ -71,6 +71,7 @@ TransformListener::TransformListener(tf2::BufferCore & buffer, bool spin_thread,
 TransformListener::~TransformListener()
 {
   if (spin_thread_) {
+    running_.store(false);
```

Comment:
I find it very surprising that this pattern is needed, as it is (essentially) the same as what executor_->cancel() should be doing under the hood.  I'm going to see if I can reproduce against `jammy`/`rolling`.